datastore/src/main/java/org/jboss/pnc/datastore/repositories/DeliverableAnalyzerLabelEntryRepositoryImpl.java

```diff
@@ -41,4 +47,27 @@ public DeliverableAnalyzerLabelEntry save(DeliverableAnalyzerLabelEntry entity)
         }
         return super.save(entity);
     }
+
+    @Override
+    public Integer getLatestChangeOrderOfReport(Base32LongID id) {
+        CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+        CriteriaQuery<Integer> query = cb.createQuery(Integer.class);
+
+        Root<DeliverableAnalyzerLabelEntry> deliverableAnalyzerReportsLabelHistory = query
+                .from(DeliverableAnalyzerLabelEntry.class);
+
+        query.select(cb.max(deliverableAnalyzerReportsLabelHistory.get(DeliverableAnalyzerLabelEntry_.changeOrder)));
+        query.where(
+                cb.equal(
+                        deliverableAnalyzerReportsLabelHistory.get(DeliverableAnalyzerLabelEntry_.report)
+                                .get(DeliverableAnalyzerReport_.id),
+                        id));
+
+        try {
+            return entityManager.createQuery(query).getSingleResult();
+        } catch (NoResultException ex) {
+            // In case the label history is empty, return the starting orderId
+            return 1;
```

Comment:
I think it should be correct this way: we want to return the latest change order for this deliverable analyzer report, e.g. we have report with id=42 with no changes so far. Then it returns us `1`. Then, if someone changes some label assigned to this report with id=42, the method would return us `1`. After 4242 changes, it would return us `4242`.

The only question is whether we should start at `1` (also `DatabaseDataInitializer` works in this manner) or `0`.